typing/typedecl.ml

```diff
@@ -549,6 +550,15 @@ let rec check_constraints_rec env loc visited ty =
           raise (Error(loc, Constraint_failed (env, err)))
       end;
       List.iter (check_constraints_rec env loc visited) args
+  | Tfunctor (_, id, (p, fl), ty) ->
+      List.iter (fun (_, t) -> check_constraints_rec env loc visited t) fl;
+      let mty = !Ctype.modtype_of_package env loc p fl in
+      let id' = Ident.(create_scoped ~scope:lowest_scope (Unscoped.name id)) in
+      let env = Env.add_module id' Mp_present mty env in
+      let ty = Option.value ~default:ty
+          (Ctype.instance_funct ~id_in:(Ident.of_unscoped id)
+              ~p_out:(Pident id') ~fixed:false ty) in
```

Comment:
"Open this Tfunctor and give me an updated environment and the right-hand-side type" is an operation that comes up fairly often in this patchset. Could you make it a helper function somewhere so that the code would be more pleasant?

Reply:
```ocaml
val open_tfunctor : env -> Ident.Unscoped.t * pack * ty -> env * Ident.t * ty
```

(Here we don't need the `Ident.t`, but my gut feeling is that in some other cases we do. @samsa1 thinks we can do without.)